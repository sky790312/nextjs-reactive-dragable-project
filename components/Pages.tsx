import React from 'react'
import styled from 'styled-components'
import Page from '@/components/Page'
import { useSelector, shallowEqual } from 'react-redux'
import { RootState } from '@/store/index'

const Pages: React.FC = () => {
  const pages = useSelector(
    (state: RootState) => state.page.pages,
    shallowEqual,
  )

  return (
    <PagesWrapper>
      <h4>Pages</h4>
      {pages.map(page => (
        <Page key={page.id} page={page} />
      ))}
    </PagesWrapper>
  )
}

export default Pages

const PagesWrapper = styled.div`
  border-bottom: 1px solid;
  padding-bottom: 16px;
`
