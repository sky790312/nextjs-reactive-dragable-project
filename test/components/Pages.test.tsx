import React from 'react'
import { render, withProviders } from '@/test/testUtils'
import Pages from '@/components/Pages'

describe('Pages component', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(withProviders(<Pages />), {})
    expect(asFragment()).toMatchSnapshot()
  })
})
