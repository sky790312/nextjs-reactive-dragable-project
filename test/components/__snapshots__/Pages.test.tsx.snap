// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Pages__PagesWrapper-hol3qz-0 fQzYNz"
  >
    <h4>
      Pages
    </h4>
  </div>
</DocumentFragment>
`;
